src/ui/common/src/components/pages/function/id/index.tsx

```diff
@@ -0,0 +1,409 @@
+import { faChevronRight } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { Link, List, ListItem } from '@mui/material';
+import Accordion from '@mui/material/Accordion';
+import AccordionDetails from '@mui/material/AccordionDetails';
+import AccordionSummary from '@mui/material/AccordionSummary';
+import Box from '@mui/material/Box';
+import Typography from '@mui/material/Typography';
+import { BlobReader, TextWriter, ZipReader } from '@zip.js/zip.js';
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { Link as RouterLink, useNavigate, useParams } from 'react-router-dom';
+
+import DefaultLayout from '../../../../components/layouts/default';
+import LogViewer from '../../../../components/LogViewer';
+import MultiFileViewer from '../../../../components/MultiFileViewer';
+import { boolArtifactNodeIcon } from '../../../../components/workflows/nodes/BoolArtifactNode';
+import { dictArtifactNodeIcon } from '../../../../components/workflows/nodes/DictArtifactNode';
+import { imageArtifactNodeIcon } from '../../../../components/workflows/nodes/ImageArtifactNode';
+import { jsonArtifactNodeIcon } from '../../../../components/workflows/nodes/JsonArtifactNode';
+import { numericArtifactNodeIcon } from '../../../../components/workflows/nodes/NumericArtifactNode';
+import { stringArtifactNodeIcon } from '../../../../components/workflows/nodes/StringArtifactNode';
+import { tableArtifactNodeIcon } from '../../../../components/workflows/nodes/TableArtifactNode';
+import {
+  handleGetArtifactResults,
+  handleGetOperatorResults,
+  handleGetWorkflow,
+  selectResultIdx,
+} from '../../../../reducers/workflow';
+import { AppDispatch, RootState } from '../../../../stores/store';
+import { ArtifactType } from '../../../../utils/artifacts';
+import UserProfile from '../../../../utils/auth';
+import { getPathPrefix } from '../../../../utils/getPathPrefix';
+import { exportFunction } from '../../../../utils/operators';
+import { LoadingStatusEnum } from '../../../../utils/shared';
+import DetailsPageHeader from '../../components/DetailsPageHeader';
+import { LayoutProps } from '../../types';
+
+type FunctionDetailsPageProps = {
+  user: UserProfile;
+  Layout?: React.FC<LayoutProps>;
+  maxRenderSize?: number;
+};
+
+const listStyle = {
+  width: '100%',
+  maxWidth: 360,
+  bgcolor: 'background.paper',
+};
+
+// Checked with file size=313285391 and handles that smoothly once loaded. However, takes a while to load.
+const FunctionDetailsPage: React.FC<FunctionDetailsPageProps> = ({
```

Comment:
Should this be named OperatorDetailsPage instead of FunctionDetailsPage? It's the title of your PR :) Either one should be fine, but our code base has more usage of "Operator" than "Function"  in terms of naming.

cc @vsreekanti and @likawind for second opinion here.

Reply:
Yeah..If this `index.tsx` is what we will use for `operator/{id}` url we should call it `OperatorDetailPage` and rename the directory as well

Reply:
Agreed that operator is more general, and we should be consistent.